packages/firebase_remote_config/firebase_remote_config/ios/Classes/FLTFirebaseRemoteConfigPlugin.m

```diff
@@ -0,0 +1,266 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Firebase/Firebase.h>
+#import <firebase_core/FLTFirebasePluginRegistry.h>
+
+#import "FLTFirebaseRemoteConfigPlugin.h"
+#import "FLTFirebaseRemoteConfigUtils.h"
+
+NSString *const kFirebaseRemoteConfigChannelName = @"plugins.flutter.io/firebase_remote_config";
+
+@interface FLTFirebaseRemoteConfigPlugin ()
+@property(nonatomic, retain) FlutterMethodChannel *channel;
+@end
+
+@implementation FLTFirebaseRemoteConfigPlugin
+
++ (instancetype)sharedInstance {
+  static dispatch_once_t onceToken;
+  static FLTFirebaseRemoteConfigPlugin *instance;
+
+  dispatch_once(&onceToken, ^{
+    instance = [[FLTFirebaseRemoteConfigPlugin alloc] init];
+    [[FLTFirebasePluginRegistry sharedInstance] registerFirebasePlugin:instance];
+  });
+
+  return instance;
+}
+
+- (instancetype)init {
+  self = [super init];
+  return self;
+}
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FlutterMethodChannel *channel =
+      [FlutterMethodChannel methodChannelWithName:kFirebaseRemoteConfigChannelName
+                                  binaryMessenger:[registrar messenger]];
+  FLTFirebaseRemoteConfigPlugin *instance = [FLTFirebaseRemoteConfigPlugin sharedInstance];
+
+  [registrar addMethodCallDelegate:instance channel:channel];
+
+  SEL sel = NSSelectorFromString(@"registerLibrary:withVersion:");
+  if ([FIRApp respondsToSelector:sel]) {
+    [FIRApp performSelector:sel withObject:LIBRARY_NAME withObject:LIBRARY_VERSION];
+  }
+}
+
+- (void)detachFromEngineForRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
```

Comment:
This will never be called because the plugin is not calling `publish`, per the API docs:

```
You will only receive this method if you registered your plugin instance with the `FlutterEngine` via `-[FlutterPluginRegistry publish:]`.
```